packages/teuchos/numerics/src/Teuchos_SerialDenseMatrix.hpp

```diff
@@ -537,6 +543,40 @@ SerialDenseMatrix<OrdinalType, ScalarType>::~SerialDenseMatrix()
 //  Shape methods
 //----------------------------------------------------------------------------------------------------
 
+template<typename OrdinalType, typename ScalarType> void
+SerialDenseMatrix<OrdinalType, ScalarType>::swap(
```

Comment:
Isn't there also an ownership flag, that governs whether the Matrix/Vector owns the data or just views them?  It could be bad if the Matrix thinks it's supposed to deallocate, but it really should not.

Reply:
Good point.  I have some additional mods from Mike Eldred I plan to include in the PR (along with unit tests) and was wondering how I refresh the PR?  Will additional commits to the associated branch trigger a refresh, or is there another step required?

Reply:
If you just push commits to the PR's branch, then that will "trigger a refresh."  You'll need new review approvals before GitHub permits a merge.  However, if you want to change Git history in any way, you'll need to create a new branch and PR.

Reply:
@mhoemmen I think the ownership flag is valuesCopied_, which I think is being used correctly to establish ownership, etc.  I just pushed the new changes which add some unit testing.  FWIW, I had to enable complex to get these activated and wasn't sure if this is desirable (though it appears consistent with other unit testing).

Reply:
What does "these" mean in "I had to enable complex to get these activated"?

PR tests disable complex by default.  There is an outstanding issue to fix that, since Sierra enables complex.

Reply:
Sorry Mark, I should have written, "... new changes which add some **unit tests**.  FWIW, I had to enable complex to get these ..." where now it's more clear that the new unit tests are the antecedent to the ambiguous pronoun.  I was surprised that the many unit tests are only active under builds with complex types enabled.  I couldn't find much evidence for these (serial dense class unit tests) on the Trilinos CDash dashboard either.

Reply:
@rhoope Thanks for explaining!  Yes, it is unfortunate that people decided just to disable tests instead of splitting them up.  Would you mind if I file an issue about that?

Reply:
I think a new issue to capture better test exposure this would be very good, especially for places the tests exist but have been masked for whatever reason.  It will probably feed into a larger gap analysis that @kddevin may have somewhere.  Thanks.

Reply:
@mhoemmen I was just curious if there was anything remaining for this PR to go forward?